File: ExtraWork/src/ticTacToe/Board.java
package ticTacToe;

public class Board {
    private char[][] board = new char[3][3];

    public Board() {
        for (int i = 0; i < 3; i++) {
            for (int j = 0; j < 3; j++) {
                board[i][j] = ' ';
            }
        }
    }

    public boolean makeMove(int row, int column, char input) {
        if (row < 0 || row >= 3 || column < 0 || column >= 3) {
            throw new IllegalArgumentException("Invalid position! Choose between 0 and 2.");
        }
        if (board[row][column] != ' ') {
            throw new IllegalArgumentException("This position is already occupied!");
        }
        board[row][column] = input;
        return true;
    }

    public String checkWinner() {
        for (int i = 0; i < 3; i++) {
            if (board[i][0] != ' ' && board[i][0] == board[i][1] && board[i][1] == board[i][2]) {
                return "Player " + board[i][0] + " wins!";
            }
            if (board[0][i] != ' ' && board[0][i] == board[1][i] && board[1][i] == board[2][i]) {
                return "Player " + board[0][i] + " wins!";
            }
        }

        if (board[0][0] != ' ' && board[0][0] == board[1][1] && board[1][1] == board[2][2]) {
            return "Player " + board[0][0] + " wins!";
        }
        if (board[0][2] != ' ' && board[0][2] == board[1][1] && board[1][1] == board[2][0]) {
            return "Player " + board[0][2] + " wins!";
        }



        return "It's a draw!";
    }

    public String getBoardAsString() {
        StringBuilder boardString = new StringBuilder();
        for (int row = 0; row < 3; row++) {
            for (int col = 0; col < 3; col++) {
                boardString.append(board[row][col] == ' ' ? "_" : board[row][col]);
                if (col < 2) {
                    boardString.append(" | ");
                }
            }
            boardString.append("\n");
        }
        return boardString.toString();
    }
}

File: ExtraWork/src/ticTacToe/board.py
class Board:
    def __init__(self):
        self.board = [[' ' for _ in range(3)] for _ in range(3)]

    def make_move(self, row, column, player):
        if row < 0 or row >= 3 or column < 0 or column >= 3:
            raise ValueError("Invalid position! Choose between 0 and 2.")
        if self.board[row][column] != ' ':
            raise ValueError("This position is already occupied!")
        self.board[row][column] = player
        return True

    def check_winner(self):
        for i in range(3):
            if self.board[i][0] != ' ' and self.board[i][0] == self.board[i][1] == self.board[i][2]:
                return f"Player {self.board[i][0]} wins!"
            if self.board[0][i] != ' ' and self.board[0][i] == self.board[1][i] == self.board[2][i]:
                return f"Player {self.board[0][i]} wins!"

        if self.board[0][0] != ' ' and self.board[0][0] == self.board[1][1] == self.board[2][2]:
            return f"Player {self.board[0][0]} wins!"
        if self.board[0][2] != ' ' and self.board[0][2] == self.board[1][1] == self.board[2][0]:
            return f"Player {self.board[0][2]} wins!"

        return "It's a draw!"

    def get_board_as_string(self):
        board_string = "\n".join(" | ".join('_' if cell == ' ' else cell for cell in row) for row in self.board)
        return board_string
